code/modules/integrated_electronics/core/printer.dm

```diff
@@ -175,6 +205,41 @@
 		metal -= cost
 		new build_type(get_turf(loc))
 
+	if(href_list["print"])
+		switch(href_list["print"])
+			if("load")
+				program = input("Put your code there:", "loading", null, null)
```

Comment:
This piece is probably gonna be the most contentious change.  I'm really against adding a method to copypaste a code to build a circuit design, since it allows for importing a design that was built outside of the current round, by someone who may not have actually designed the circuit and didn't build it themselves, which is one of the things which ruined NTSL for me (as well as the security issues).  Speaking of security issues, this doesn't appear to be sanitized at all, and even if it was, I'm still paranoid at how this could be used to break things.

Reply:
Yes, that's a point. And that's a point to ignore refs , when it is scanning assembly data.
Because it's a way to master your creations for many rounds. Test it, make it more optimised and perfect.
About sanitize,there is sanity check of program.